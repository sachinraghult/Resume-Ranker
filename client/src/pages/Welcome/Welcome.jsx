import { Fragment, useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import React from "react";
import styles from "./Welcome.module.css";
import LoadingScreen from "react-loading-screen";
import spinner from "../../components/LoadingScreen/load.gif";

import { Url } from "../../enums";

const Welcome = () => {

  const navigate = useNavigate();

  const handleNewUser = () => {
    navigate(Url.SELECTION, { replace: true });
  };

  const handleOldUser = () => {
    navigate(Url.LOGIN, { replace: true });
  };

  const [isLoading,setIsLoading] = useState(true);

    useEffect(() => {
        setTimeout(() => {
            setIsLoading(false);
        },3000);
    },[]);

  return (
    <Fragment>

      {isLoading?(
        <LoadingScreen
        loading={true}
        bgColor = "#f1f1f1"
        logoSrc={spinner}
        />
      ):(
        <div className={styles.loginWrapper}>
        <div className={styles.loginLeft}>
          <h3 className={styles.loginLogo}>Job Finder</h3>
          <span className={styles.loginDesc}>Enter your details...</span>
        </div>
        <div className={styles.welcomeRight}>
          <button
            data-testid="handleNewUser"
            onClick={handleNewUser}
            style={{ backgroundColor: "#4B5D67" }}
          >
            New User
          </button>
          <button data-testid="handleOldUser" onClick={handleOldUser}>
            Existing User
          </button>
        </div>
      </div>
      )}
      {/* { user.value === 'Register' && } */}
      {/* { user.value === 'Login' && <Login/>} */}
    </Fragment>
  );
};

export default Welcome;
