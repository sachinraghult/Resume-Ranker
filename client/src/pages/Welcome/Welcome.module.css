.loginWrapper {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0%;
  font-family: "Roboto Slab", serif;
}

@media screen and (max-width: 900px) {
  .loginWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    /* background-color: blue;  */
  }
}

.loginLeft {
  /* background-color: aqua;  */
  display: flex;
  padding-top: 10px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  margin: 0px;
  height: 100vh;
  background-image: url("../../../public/assests/img1.svg");
  background-repeat: no-repeat;
  object-fit: fill;
}

.loginRight {
  /* background-color: aqua; */
  width: 100%;
  display: flex;
  margin-left: 0px;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  /* background-color: aqua;  */
}

.orContainer {
  margin-top: 20px;
}
.loginLogo {
  font-size: 50px;
  font-weight: 800;
  color: rgb(5, 28, 103);
  margin-top: 250px;
}

.RegisterBox {
  height: 600px;
  margin-right: 100px;
  margin-top: 30px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

@media screen and (max-width: 900px) {
  .RegisterBox {
    margin: auto;
    margin-top: 20px;
  }
  .loginLeft {
    height: 30vh;
  }
}
@media screen and (max-width: 310px) {
  .RegisterBox {
    margin: 0px;
    margin-top: 20px;
    padding: 0%;
  }
}

.RegisterBox span {
  text-align: center;
  margin-top: 10px;
}

.RegisterBox label {
  font-size: 20px;
  margin-top: 15px;
}

.loginInput {
  width: 92.5%;
  height: 45px;
  border-radius: 4px;
  border: 1px solid gray;
  font-size: 18px;
  padding-left: 20px;
}

.loginInput:focus {
  outline: none;
}

.loginButton {
  margin-top: 5px;
  width: 100%;
  height: 40px;
  border-radius: 4px;
  border: 1px solid gray;
  font-size: 20px;
  background-color: rgb(0, 0, 0);
  color: white;
  cursor: pointer;
  margin-top: 30px;
}
.loginButton:hover {
  background-color: blue;
}
.googleButton {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 50px;
  border-radius: 4px;
  border: 1px solid gray;
  font-size: 20px;
  background-color: #fff;
  color: black;
  cursor: pointer;
}

.googleButton img {
  margin-right: 20px;
}

.loginForgot {
  text-align: center;
  color: blue;
}

.loginRegisterButton {
  width: 80%;
  height: 40px;
  border-radius: 10px;
  border: 1px solid gray;
  font-size: 15px;
  padding: 0 20px;
  background-color: rgb(39, 83, 0);
  color: white;
  cursor: pointer;
}

.formtop {
  font-size: 25px;
}

@media screen and (max-width: 330px) {
  .loginButton {
    width: 100%;
    height: 50px;
  }
  .googleButton {
    width: 100%;
    height: 50px;
  }
}

.welcomeRight {
  display: flex;
  flex-direction: column;
  /* align-items: center; */
  justify-content: center;
  width: 100%;
  height: 100%;
}

.welcomeRight button {
  margin-left: 20%;
  width: 40%;
  height: 50px;
  background-color: #00bcd4;
  border-radius: 5px;
  border: none;
  color: #ffffff;
  font-size: 20px;
  font-family: "Roboto Slab", serif;
  cursor: pointer;
  margin-top: 30px;
}

.welcomeRight button:hover {
  background-color: #0097a7;
  transform: scale(1.2, 1.2);
  transition-delay: 0.2s;
}

@media screen and (max-width: 900px) {
  .welcomeRight {
    align-items: center;
  }
  .welcomeRight button {
    margin-left: 0%;
  }
}

@media screen and (max-width: 450px) {
  .welcomeRight {
    margin-top: 30px;
    align-items: center;
  }
  .welcomeRight button {
    width: 70%;
    height: 50px;
    background-color: #00bcd4;
    border-radius: 5px;
    border: none;
    color: #ffffff;
    font-size: 20px;
    font-family: "Roboto Slab", serif;
    cursor: pointer;
    margin-top: 30px;
    margin-left: 0%;
  }
}
@media screen and (max-width: 320px) {
  .welcomeRight {
    margin-top: 50px;
    width: 50%;
  }
  .welcomeRight button {
    width: 100%;
    height: 50px;
    background-color: #00bcd4;
    border-radius: 5px;
    border: none;
    color: #ffffff;
    font-size: 20px;
    font-family: "Roboto Slab", serif;
    cursor: pointer;
    margin-top: 30px;
  }
}
