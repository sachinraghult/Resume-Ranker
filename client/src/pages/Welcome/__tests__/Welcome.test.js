import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import Welcome from "../Welcome";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <Welcome />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Welcome />
    </BrowserRouter>
  );
  fireEvent.click(getByTestId("handleNewUser"));
  fireEvent.click(getByTestId("handleOldUser"));
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Welcome />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
