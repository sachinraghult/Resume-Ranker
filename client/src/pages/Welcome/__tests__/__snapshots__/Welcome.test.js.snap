// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loginWrapper"
>
  <div
    className="loginLeft"
  >
    <h3
      className="loginLogo"
    >
      Job Finder
    </h3>
    <span
      className="loginDesc"
    >
      Enter your details...
    </span>
  </div>
  <div
    className="welcomeRight"
  >
    <button
      data-testid="handleNewUser"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#4B5D67",
        }
      }
    >
      New User
    </button>
    <button
      data-testid="handleOldUser"
      onClick={[Function]}
    >
      Existing User
    </button>
  </div>
</div>
`;
